### PR TITLE
Got all of the verbs working for Customers. There are 3 GETs, a Post,…

### LINKED ISSUE #1: Allow developers to access Customer resource
Verbs to be supported

- [x] GET
- [x] POST
- [x] PUT

User should be able to GET a list of customers, and GET a single customer.

### PR DESCRIPTION
**Pull request by joechesney**
… and a PUT

# Description
Got all of the verbs working for Customers. There are 3 GETs, a Post, and a Put.

## Number of Fixes
Fixes #1 

## Related Ticket(s)
#1 

## Problem to Solve
Gives users access to the Customers Table

## Expected Behavior
Should correctly Get all customers, or Get a single customer, or Get all customers that have not placed an order, or Post a new customer to the Customers table, or Put new property values onto an existing customer.

## Steps to Test Solution

1. To test the GET allCustomers function, go to `http://127.0.0.1:8080/api/v1/customers` and there should be all of the customers listed, which I believe is 25 total customers. 
2. To test the GET oneCustomer function, go to `http://127.0.0.1:8080/api/v1/customers/15` (or replace the number 15 in the url with any number between 0 and 24, and there should be one customer listed, and their customer_id should match the number you typed into the url.
3. ALSO go to `routes/customers/index.js` and make sure the route for customers look s like this:
`productsIndexRouter.use("/customers", require('./customers'));`
4. THEN go to `routes/products/customers.js` and make sure the routes look like this:
`customersRouter.get('/:id', getOneCustomer);`

`customersRouter.put('/:id', updateCustomer);`

`customersRouter.get('/', getAllCustomers);`

`customersRouter.post('/', postNewCustomer);`

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

### CODE REVIEW

File: routes/products/customers.js [hunk 1]
```diff
@@ -5,11 +5,12 @@ const appRoot = process.cwd();
...
 const customersRouter = Router();
 
-const { getAllCustomers, getOneCustomer } = require(appRoot + '/controllers/products/customersCtrl');
+const { getAllCustomers, getOneCustomer, postNewCustomer, updateCustomer } = require(appRoot + '/controllers/products/customersCtrl');
 
 //TODO: routes & their corresponding controllers are enumerated here
-customersRouter.get('/customers/:id', getOneCustomer);
-customersRouter.get('/customers', getAllCustomers);
-
+customersRouter.get('/:id', getOneCustomer);
```
Comment (jordan-castelloe) [inline routes/products/customers.js:11]: This didn't work for me, but I changed it to ``/customers/:id`` and that made everything work!

**Review by jordan-castelloe — CHANGES_REQUESTED** [review]
Add ``/customers`` to your routes in ``routes/products/customers.js`` or ``routes/products/index.js``! I added that on my local version of your branch and tested again and everything else worked like a charm.

**Comment by jordan-castelloe** [comment]
Also, for the next person who tests this branch, here's the JSON I used for POST and PUT so you don't have to write it again.
`` {
"first_name": "Nicolas",
"last_name": "Cage",
"account_creation_date": "2018-03-22T15:52:23.254Z",
"street_address": "123 Awesome Street",
"city": "Crab Apple",
"state": "North Carolina",
"postal_code": "100000",
"phone_number": "(042) 494-3771 x385"
} ``

**Comment by jordan-castelloe** [comment]
Also, @joechesney: it looks like you checked a box that there's new documentation in this pr that I should review, but I wasn't sure where to look. Happy to look at it, just point me in the right direction!

File: routes/products/customers.js (hunk 1 above)
Comment (jordan-castelloe) [reply routes/products/customers.js:11]: OH okay wait I just looked at David's PR and he added it in the index file for his sub-folder! I didn't even think of that, but that works too. That might have been what you were doing and just forgot to commit the index file with the ``/customers`` route in it?

Comment (DavidLarsKetch) [reply routes/products/customers.js:11]: Yeah @kenziebottoms has been doing that I really liked it, so stole it from them.

Comment (kenziebottoms) [reply routes/products/customers.js:11]: :star2:

**Comment by DavidLarsKetch** [comment]
Don't forget to test [http://localhost:8080/api/v1/customers?active=false](http://localhost:8080/api/v1/customers?active=false)!

File: routes/products/customers.js (hunk 1 above)
Comment (DavidLarsKetch) [reply routes/products/customers.js:11]: Ditto what @jordan-castelloe. Everything works :100: once you throw `/customers` into `routes/products`

File: controllers/products/customersCtrl.js [hunk 2]
```diff
@@ -35,8 +35,35 @@ const getOneCustomer = (req, res, next) =>{
   });
 };
 
+const postNewCustomer = (req, res, next) =>{
+  console.log('req body',req.body);
```
Comment (DavidLarsKetch) [inline controllers/products/customersCtrl.js]: Get rid of dat stank

**Review by DavidLarsKetch — CHANGES_REQUESTED** [review]
Same what @jordan-castelloe said and that one left in `console.log` in `customersCtrl`

**Comment by kenziebottoms** [comment]
testing

**Review by kenziebottoms — APPROVED** [review]

**Pull request merged**